### PR TITLE
Feat/improvements btc only

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- move logic which decides which coin is visible and which is not, which is disabled and which is not to selector
- use selector in add account modal, hide `Enable more coins` button when all mainnets are enabled
- in add account modal, open bitcoin type select directly on btc-only fw in case of only btc is enabled
- show `Enable more coins` button next to `Assets` part on dashboard only when there are mainnets available to be activated
- removed all altcoin T1/TR homescreen images
- lower case `Active coins`, also done in crowdin
- remove add account button from account detail when no coins enabled, the opening of modal was broken anyway
- refactor <MoreCoins /> and use supported networks selector there. It is not perfect but better than before.
- fix coin settings section, show btc-only fw banner when only btc is enabled
  - there was a problem when you had enabled coin from different device which is unsupported on the second device, it did not offer btc-only fw when "only" btc was enabled
- translations are also changed in crowdin
- testnet coins padding in onboarding coins
- different text for btc-only and regular fw in onboarding coins

## Related Issue

Four bottom parts of [#33](https://github.com/trezor/trezor-suite-private/issues/33)
it is now better than task requirements

## Screenshots:
Screenshot 2023-08-16 at 15 43 35
Screenshot 2023-08-16 at 15 43 51
Screenshot 2023-08-16 at 15 46 03
on open
Screenshot 2023-08-16 at 15 45 25
Screenshot 2023-08-16 at 16 03 38

### CODE REVIEW

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx [hunk 1]
```diff
@@ -181,7 +189,7 @@ export const AddAccount = ({ device, onCancel, symbol, noRedirect }: AddAccountP
                 selectedNetworks={selectedNetworks}
                 handleNetworkSelection={handleNetworkSelection}
             />
-            {!selectedNetworkEnabled && hasDisabledNetworks && (
+            {hasNetworksToEnable && !selectedNetworkEnabled && hasDisabledNetworks && (
```
Comment (matejkriz) [inline packages/suite/src/components/suite/modals/AddAccount/index.tsx]: check if `hasDisabledNetworks` always true if `!!hasNetworksToEnable`

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx [hunk 2]
```diff
@@ -77,21 +81,37 @@ export const AddAccount = ({ device, onCancel, symbol, noRedirect }: AddAccountP
...
             }
-        } else {
-            selectNetwork(undefined);
+        },
+        [networkPinned, selectedNetwork?.symbol],
+    );
+
+    // call just once on open, if btc-only fw and only btc is enabled, show account type selection
+    useEffect(() => {
+        console.log(supportedNetworks, allTestnetNetworksDisabled, device.firmwareType);
```
Comment (matejkriz) [inline packages/suite/src/components/suite/modals/AddAccount/index.tsx]: leftover

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx [hunk 3]
```diff
@@ -77,21 +81,37 @@ export const AddAccount = ({ device, onCancel, symbol, noRedirect }: AddAccountP
         (n: Network) => !(n.symbol === 'regtest' && !debug.showDebugMenu),
     );
 
-    const handleNetworkSelection = (symbol?: Network['symbol']) => {
-        if (symbol) {
-            const networkToSelect = NETWORKS.find(n => n.symbol === symbol);
+    const handleNetworkSelection = useCallback(
```
Comment (matejkriz) [inline packages/suite/src/components/suite/modals/AddAccount/index.tsx]: useCallback not needed

Comment (tomasklim) [reply packages/suite/src/components/suite/modals/AddAccount/index.tsx]: 02f7f6de87283cd2290b0e3c80f7028c38846769

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx (hunk 2 above)
Comment (tomasklim) [reply packages/suite/src/components/suite/modals/AddAccount/index.tsx]: 02f7f6de87283cd2290b0e3c80f7028c38846769

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx (hunk 3 above)
Comment (matejkriz) [reply packages/suite/src/components/suite/modals/AddAccount/index.tsx]: also remove import of useCallback as it is no more used in this file

File: packages/suite/src/reducers/suite/suiteReducer.ts [hunk 4]
```diff
@@ -314,4 +316,33 @@ export const selectIsDiscoveryAuthConfirmationRequired = (
...
 };
 
+export const selectSupportedNetworks = (state: SuiteRootState) => {
+    const device = selectDevice(state);
+    const deviceModelInternal = device?.features?.internal_model;
+    const firmwareVersion = getFirmwareVersion(device);
+
+    return Object.entries(networks)
+        .map(([symbol, network]) => {
+            // @ts-expect-error TODO resolve why support is not there according to types
```
Comment (matejkriz) [inline packages/suite/src/reducers/suite/suiteReducer.ts]: `support` is only on ada/tada and networks is defined as const so the type is union of all network types, not merge... 

so something like 
`const support = 'support' in network ? network.support : undefined;` would work...

**Comment by tomasklim** [comment]
@trezor/qa 
- no altcoin homescreens on T1/TR
- no add account button in account when all coins disabled
- btc-only fw banner in coins section visible when btc-only fw or when only btc/test/regtest enabled (you can enable cardano on TT, then connect T1 and test it, on production you wont see the banner, here you will)
- Enable more coins not visible on dashboard when all mainnets are enabled (applies for both fw, do the test with cardano and T1)
- Add account modal - no "Enable more coins" button when all mainnets are enabled (applies for both fw, do test with cardano and T1)
- When only btc is enabled on btc-only fw and you open add account, it should preselect btc 
- check - coin enable/disable, custom backend settings, mobile view coins
- different text in onboarding in coins section

File: packages/suite/src/components/wallet/AccountException/DiscoveryEmpty.tsx
Comment (komret) [inline packages/suite/src/components/wallet/AccountException/DiscoveryEmpty.tsx:1]: I don't know about this change, it worked... But I guess it makes sense as it eliminates an unnecessary choice. However, it is still possible to add account via the Accounts bar and Dashboard (the latter seems to be an oversight).

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx [hunk 5]
```diff
@@ -37,10 +40,27 @@ export const AddAccount = ({ device, onCancel, symbol, noRedirect }: AddAccountP
...
+    const testnetSymbols = testnets.map(testnet => testnet.symbol);
+
+    const supportedNetworks = useSelector(selectSupportedNetworks);
+    const supportedMainnetNetworks = supportedNetworks.filter(network =>
+        mainnetSymbols.includes(network),
+    );
+
+    const hasMainnetNetworksToEnable = !!supportedMainnetNetworks.filter(
+        network => !enabledNetworksSymbols.includes(network),
+    ).length;
```
Comment (komret) [inline packages/suite/src/components/suite/modals/AddAccount/index.tsx]:
```suggestion
    const hasMainnetNetworksToEnable = supportedMainnetNetworks.some(
        network => !enabledNetworksSymbols.includes(network),
    );
```

File: packages/suite/src/components/suite/modals/AddAccount/index.tsx [hunk 6]
```diff
@@ -37,10 +40,27 @@ export const AddAccount = ({ device, onCancel, symbol, noRedirect }: AddAccountP
...
+
+    const allTestnetNetworksDisabled = !supportedNetworks.filter(
+        network => testnetSymbols.includes(network) && enabledNetworksSymbols.includes(network),
+    ).length;
```
Comment (komret) [inline packages/suite/src/components/suite/modals/AddAccount/index.tsx]:
```suggestion
    const allTestnetNetworksDisabled = !supportedNetworks.some(
        network => testnetSymbols.includes(network) && enabledNetworksSymbols.includes(network),
    );
```

File: packages/suite/src/components/suite/CoinsGroup/CoinsList.tsx
Comment (komret) [inline packages/suite/src/components/suite/CoinsGroup/CoinsList.tsx:1]: There is some repeated code from `selectSupportedNetworks`. `getCoinUnavailabilityMessage` can now only accept 'update-required' as an argument, could it be a bug?

**Review by komret — CHANGES_REQUESTED** [review]

**Review by komret — APPROVED** [review]

**Comment by bosomt** [comment]
**QA NOK**


- app remembers that Ethereum was enabled and even performs discovery with BTC only FW 
- wallet was not remembered 

image

image

image

image

**Info:**
 - Suite version: desktop 23.9.0 (27563c630b9ff177bc2f716eb5b23ea432e187dc)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/23.9.0 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T1B1 1.12.1 bitcoin-only (revision 1eb0eb9d91b092e571aac63db4ebff2a07fd8a1f)
 - Transport: BridgeTransport 2.0.33

**Comment by bosomt** [comment]
Add account button is still available even when all coins are disabled


image


> @trezor/qa
> 
>     * no add account button in account when all coins disabled
>

**Comment by tomasklim** [comment]
> Add account button is still available even when all coins are disabled
> 
> image
> > @trezor/qa
> > ```
> > * no add account button in account when all coins disabled
> > ```

not true, you probably did this, if yes, please create an issue
https://github.com/trezor/trezor-suite/assets/33235762/a957b260-e132-4895-99b5-8c6607356b8f

**Comment by bosomt** [comment]
**QA OK**

Separate issue, i will replicate and create new one.

**Comment by bosomt** [comment]
It looks like same issue that i reported some time ago
https://github.com/trezor/trezor-suite/issues/9115

**Comment by bosomt** [comment]
**QA OK**

retest with firmware-T2B1-btconly-2.6.1-364c1ce12

**Pull request merged**